linfa-svm/examples/winequality.rs

```diff
@@ -0,0 +1,85 @@
+#[macro_use]
+extern crate ndarray;
+
+use std::error::Error;
+use std::fs::File;
+
+use csv::ReaderBuilder;
+use flate2::read::GzDecoder;
+use ndarray::{Array2, Axis};
+use ndarray_csv::Array2Reader;
+
+use linfa::metrics::IntoConfusionMatrix;
+use linfa_kernel::Kernel;
+use linfa_svm::{SVClassify, SolverParams};
+
+fn read_array(path: &str) -> Result<Array2<f64>, Box<dyn Error>> {
+    let file = GzDecoder::new(File::open(path)?);
+    let mut reader = ReaderBuilder::new()
+        .has_headers(true)
+        .delimiter(b';')
+        .from_reader(file);
+    let array = reader.deserialize_array2_dynamic()?;
+    Ok(array)
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    // read in wine-quality dataset
+    let data = read_array("../datasets/winequality-red.csv.gz")?;
+
+    // calculate number of training and validation points
+    let npoints = data.len_of(Axis(0));
```

Comment:
It might useful to add a comment like 
```
// .csv data is two-dimensional: Axis(0) denotes y-axis (rows), Axis(1) denotes x-axis (columns)
```
to make this more clear to people that aren't familiar with the ndarray's slicing/axis conventions

Reply:
yeah you're right will improve the comments .. they were meant as a remainder for myself, but the audience is obviously different here

Reply:
okay added some comments in the example

Reply:
Looked through most recent commits, the new comments and updated fields look great